UnitySetup/UnitySetup.psm1

```diff
@@ -2254,8 +2246,18 @@ Would you like to continue? (Default: $($createPAT))"
             }
             Connect-AzAccount @connectArgs  | Out-Null
         }
-        $AZTokenRequest = Get-AzAccessToken -ResourceType Arm
-        $headers = @{ Authorization = "Bearer $($AZTokenRequest.Token)" }
+
+        $AZToken = $null
+        $azAccountsVersion = Get-ModuleVersion -ModuleName "Az.Accounts"
+        if ($azAccountsVersion -ge [version]"4.0") {
+            $AZTokenRequest = Get-AzAccessToken -AsSecureString -ResourceType Arm
+            $AZToken = [System.Runtime.InteropServices.Marshal]::PtrToStringAuto([System.Runtime.InteropServices.Marshal]::SecureStringToBSTR($AZTokenRequest.Token))
```

Comment:
Nit: in other places in this file we use the below to get a string from a secure string, not arguing it's better, just think it'd be good to be consistent.

```powershell
[System.Net.NetworkCredential]::new($null, $secureString).Password
```

Reply:
Good call - yeah better to be consistent / I didn't notice our other conversion here.  This abomination was what copilot suggested for securestring -> str conversion and I don't particularly like it.